smart-contracts/test/KeyManager/upgrades.js

```diff
@@ -0,0 +1,9 @@
+contract('KeyManager', () => {
+
+  beforeEach(async () => {
```

Comment:
Any suggestion on testing this @clemsos ?

Reply:
Why make it upgradeable in the first place? Should be easy enough to just redeploy a new one as there is no storage (except locksmith address).

Reply:
There is a possible example with the `TestPublicLockUpgraded` contract. Basically you have to import yout own contract into a new one like:

```solidity
import './KeyManager.sol';

contract KeyManagerV2 is KeyManager {
  function someNewFeature() {... }
}
```

then you can test with sth like that :

```js
const { ethers, upgrades } = require('hardhat')

const KeyManager = await ethers.getContractFactory('KeyManager')
const proxy = await upgrades.deployProxy(KeyManager, args)

const KeyManagerV2 = await ethers.getContractFactory('KeyManagerV2')

await upgrades.upgradeProxy(proxy.address, KeyManagerV2)
```

Here is an example with a similar pattern https://github.com/unlock-protocol/unlock/blob/0d8e0a5d93513c5d6a52a7f5f5340a0aef603e33/smart-contracts/test/Lock/upgrades/v11.js#L91

Reply:
Perfect! Thanks a lot!

Reply:
Done!